ConnectorRhino/ConnectorRhino/ConnectorRhinoShared/UI/MappingBindingsRhino.cs

```diff
@@ -26,41 +26,44 @@ public MappingBindingsRhino()
 
   public override MappingSelectionInfo GetSelectionInfo()
   {
-    try
+    List<RhinoObject> selection = RhinoDoc.ActiveDoc.Objects.GetSelectedObjects(false, false)?.ToList();
+    if (selection is null)
     {
-      var selection = RhinoDoc.ActiveDoc.Objects.GetSelectedObjects(false, false).ToList();
-      var result = new List<Schema>();
+      return new MappingSelectionInfo(new List<Schema>(), 0);
+    }
 
-      foreach (var obj in selection)
-      {
-        var schemas = GetObjectSchemas(obj);
+    var result = new List<Schema>();
+    foreach (var obj in selection)
+    {
+      List<Schema> schemas = GetObjectSchemas(obj);
 
-        if (!result.Any())
-        {
-          result = schemas;
-        }
-        else
+      if (!result.Any())
+      {
+        result = schemas;
+      }
+      else
+      {
+        //intersect lists
+        //TODO: if some elements already have a schema and values are different
+        //we should default to an empty schema, instead of potentially restoring the one with values
+        try
         {
-          //intersect lists
-          //TODO: if some elements already have a schema and values are different
-          //we should default to an empty schema, instead of potentially restoring the one with values
           result = result.Where(x => schemas.Any(y => y.Name == x.Name)).ToList();
         }
-
-        //incompatible selection
-        if (!result.Any())
-        {
-          return new MappingSelectionInfo(new List<Schema>(), selection.Count);
-        }
+        catch (ArgumentNullException) { }
       }
```

Comment:
Lot's in this, can schemas be null? I am not a fan of the check !Result.Any() but reading it again it seems that if we don't have anything we overwrite the original object with the result of GetObjectSchema() which appears to always return but is itself swallowing all exceptions.

Ultimately my question is whether the moving of the try exposes us to legitimate exceptions and then secondary we are just swallowing ArgumentNullException. That feels like a smell that something can be null. It seems the fix has been to catch the exception where I would argue nothing at this point should be null - it must be x or y because comparing x.Name and y.Name should not explode any.

Reply:
Still not sure about this one - I'd like to understand the empty catch and whether we can avoud the need to for t/c

Reply:
> would this argument exception be because `result` is null? or because `x.Name` was?
> 
> If we could prevent it from happening instead i think that would be preferable (adding some null checks in key places)
> 
> I'm aware this is a mappings class so i'd tread lightly before asking for too many changes 😅

`schemas` wouldn't be null, but not sure about `x.name` - similarly I'm a bit afraid to touch anything related to mappings

Reply:
I'm removing the trycatch with a null check🤞